api/http/common.go

```diff
@@ -96,6 +100,19 @@ func ValidateUUID(extID string) (err error) {
 	return nil
 }
 
+// ValidateName validates name format.
+func ValidateName(id string) error {
+	if !nameRegExp.MatchString(id) {
+		return apiutil.ErrInvalidNameFormat
+	}
+	// Names containing double underscores or double dashes are invalid due to similarity concerns.
+	if strings.Contains(id, "__") || strings.Contains(id, "--") {
+		return errors.Wrap(apiutil.ErrInvalidNameFormat, errors.New("name containing double underscores or double dashes not allowed"))
```

Comment:
Do not create a new error each time, extract `errors.New("name containing double underscores or double dashes not allowed")` to an unexported error var.